### ISSUE TITLE
CI: When installer fails  to install NVDA but does not crash NVDA log cannot be uploaded successfully and build is not interrupted.

### ISSUE THREAD

**Issue by lukaszgo1** [nvaccess/nvda#13083]
### Steps to reproduce:
1. Open a PR against NVDA repository which causes NVDA installer to hang (for an example see #13079 )

### Actual behavior:
1. There is no comment on the pr saying 'NVDA installer process timed out'
2. Log from the installer is not uploaded.
3. Build tries to continue and system tests are executed against broken NVDA installation and obviously fail.
### Expected behavior:
1. There is a message that NVDA installer failed to install in the specified time.
2. Installer log is uploaded so developers can download it from the artifacts tab and inspect to determine reason for the failure.
3. Build is interrupted since tests cannot be executed if NVDA is not installed.
#### NVDA installed/portable/running from source:
Affects installer.
#### NVDA version:
Latest master as of November 22-th
#### Windows version:
I believe AppVeyor worker images are running Server 2019
#### Name and version of other software in use when reproducing the issue:
None
#### Does the issue still occur after restarting your computer?
N/A
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
N/A
#### If NVDA add-ons are disabled, is your problem still occurring?
Irrelevant
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Irrelevant.

### Technical:
Points 1 and 3 are caused by the fact that by default exceptions raised by `wait-process` are not terminating - therefore script never gets into the `catch` block and does not execute this code path.
However even when the exception is raised and caught the installer log fails to upload with the following, not very helpful output:
```
Push-AppveyorArtifactInternal : Error uploading artifact to the storage: One or more errors occurred.
At C:\Program Files\AppVeyor\BuildAgent\Modules\build-worker-api\build-worker-api.psm1:209 char:2
+     Push-AppveyorArtifactInternal -FullPath $fullPath -FileName $File ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Push-AppveyorArtifactInternal], Exception
    + FullyQualifiedErrorId : System.Exception,Appveyor.BuildAgent.Api.Utils.PushAppveyorArtifactInternalCmdlet
Command executed with exception: Error uploading artifact to the storage: One or more errors occurred.
```

cc @seanbudd  I believe you've worked on AppVeyor scripts recently.
